qutip/cy/cqobjevo.pyx

```diff
@@ -224,20 +232,32 @@ cdef class CQobjEvo:
 
     cpdef complex expect(self, double t, complex[::1] vec):
         if self.super:
+            if self.shape1 != vec.shape[0]:
+                raise ValueError(f"Shape don't match: ({self.shape0}, "
+                                 f"{self.shape1}) @ ({vec.shape[0]}, 1) ")
             return self._expect_super(t, &vec[0])
         else:
+            if self.shape1 != vec.shape[0] and self.shape0 != self.shape1:
+                raise ValueError(f"Shape don't match: ({self.shape0}, "
+                                 f"{self.shape1}) @ ({vec.shape[0]}, 1) ")
             return self._expect(t, &vec[0])
 
     def overlapse(self, double t, complex[::1] oper):
```

Comment:
Should this method be called `overlap` or `overlaps`? `overlapse` is not an English word (although it is a very cool made up word). Apologies for not noticing this in earlier reviews. Maybe let's fix it in a separate PR to this one.

Reply:
Or maybe "overlapse" is a technical QM term I'm not familiar with?

Reply:
No it's a spelling error.